### DISCUSSION TITLE
Firefox can't download videos with special characters

### DISCUSSION THREAD

**Discussion by nemesiscompletion** [aclap-dev/video-downloadhelper#1149] [Help and Q&A]
### Your question

Since last update from Firefox, extensions can't download anything with special characters on the title of the page
For example, if the title of the page has : ; or . , the download just doesn't work. It just keep fails

There's image extension called Image Picka which had same problem
What the developer said is change the file name download option from  Image Picka/${pageTitle}/${name}${ext} to this
Image Picka/${pageTitle.replace(/%|\./g,'_')}/${index} - ${name.replace(/%/g,'_')}${ext}
This solved the problem and I could download the images

However, I'm not seeing any way to change the download option for Video Downloadhelper
Can you fix this so we can download files?
There are a lot of web pages that has special characters in the title, and we just can't download anything

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-CA
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "download_directory": "C:\\Users\\Klace\\dwhelper",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 8,
    "successfull_dl": 5
}

**Comment by Mattmjs** [comment]
Have you got an example to share here for other 
users to look at.

**Comment by GrampaWildWilly** [comment]
I don't work for VDH.  I'm just another user.

I was also going to ask for some examples.  In the meantime, you should use the VDH `Download As...` function in preference to the simple `Download` function.  `Download As...` offers you the possibility of manually editing the target file name before you launch the download.

In addition, certain characters are not permitted in file names on Windows.  I can't say anything about Linux or Mac since I don't use them.  But on Windows, this is the list of prohibited characters in file names:

/ \ | : * ? " &lt; &gt;

Of the characters you mention in your original post, colon is not permitted.  The others are, including comma.  Personally, I avoid the comma in file names.  Certain applications misinterpret the comma in the file name as a separator & misconstrue your file name as 2 file names, separated by the comma.  But comma is, nevertheless, valid.  Other perhaps surprising characters accepted on Windows are semicolon, exclamation point, ampersand, percent, apostrophe.  Of course, the period is accepted.  It's the separator between the file name & the file extension.  But you can include several periods in a file name, in addition to the one that sets off the file extension.  That is, if the file name has a file extension.  A file extension is not required.  Windows also accepts most accented characters in file names, like é, à, ô, ü, and many more.

So for the time being, my advice is to never use `Download` and always use `Download As...` so you can fix whatever errors there might be yourself.  You should start following this advice immediately.  That is, unless you want to wait some unknown time for @paulrouget Paul to get around to fixing this issue.

**Comment by GunnarBernsteinHH** [comment]
Yes, unfortunately this even applies to Download, where you can set the filename. The file save dialog does not even come up.

**Reply by Mattmjs** [reply]
But isn't the Download action what launches 
it right away without a dialog. I've got it set to
Download as. This offers a dialog to choose a
directory and if needed alter the file name.

I saw this problem a few months ago on Rumble. Sometime after it was fixed.
Maybe you should share an example or try smartnaming.

**Comment by Mattmjs** [comment]
It just so happens that I came across a video that
had " characters in the file name. I have _download as_  set but it never gets to the dialog.
Just immediate failure. This is a http media where I use the browser for the download. I have to see if using companion app will work.

**Comment by Mattmjs** [comment]
Using the companion app for the download,
the dialog to save comes up. So just untick 
the setting _Do not use Coapp when possible_
and it'll work.
